include/RAJA/util/PluginStrategy.hpp

```diff
@@ -9,26 +9,30 @@
 #define RAJA_PluginStrategy_HPP
 
 #include "RAJA/util/PluginContext.hpp"
+#include "RAJA/util/PluginOptions.hpp"
 #include "RAJA/util/Registry.hpp"
 
 namespace RAJA {
 namespace util {
 
-
 class PluginStrategy
 {
   public:
     PluginStrategy();
 
     virtual ~PluginStrategy() = default;
 
-    virtual void preCapture(PluginContext p);
+    virtual void init(PluginOptions p);
```

Comment:
Should this be by reference too?

Reply:
We talked about that for a little bit, PluginOptions at the moment is basically just an std::string, which is certainly expensive to copy, however if they all share the same string one plugin could modify it and then another might receive the wrong information. The default init_plugins() command will put in an empty string by default, so this copy is pretty tiny. It's definitely possible to just pass a reference, but that's a design decision we can make

Reply:
const& then?

Reply:
@DavidPoliakoff The way we've implemented support for Kokkos Tools is by including the kID in RAJA's PluginContext. If we pass that as const, do you think it would interfere with any of the tools? I don't think it should but I'm not sure.

Reply:
I'd recommend const for post but not pre. The pre-hooks often write to data, calling things const might break that flow

Reply:
Further, though, I think most tools won't provide interesting data given empty names. The kID mostly just exists to speed up some "end" call for some tools

Reply:
This is a little confusing to me, are the Kokkos Tools setting kID in the pre hook?

Reply:
Yes, they are.

Reply:
Right, the typical pattern is that the begin hook does some start event, and then adds some context into a data structure there. They overwrite the contents of kID, and then on the end event they use kID to look up that earlier context. This saves a string lookup, which is of debatable utility in my view, but it keeps tool people from yelling at me, which is the most important thing

Reply:
Ok that makes sense, 
It seems somewhat dangerous to have an unprotected kID in the context. Other plugins could use it to speedup their own post hooks. This also leaves the platform member exposed.
Is there a design that doesn't expose these variables, or only exposes them to the kokkos plugins?

Reply:
Would it work to make kID be a private mutable member of the context and give the context a private const friend accessor for KokkosPluginLoader?
That way the signatures of the plugins could all be const ref and only the kokkos plugins could access kID.

Reply:
Sounds like a good idea, I'm trying to get it working but it seems to be giving me a little trouble. I'll see what I can do!